### PR TITLE
Running RonDB key/value API server unit tests using MTR

### PR DESCRIPTION
**Pull request by smkniazi**

### CODE REVIEW

File: mysql-test/suite/rdrs/include/have_rdrs.inc [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+                                NOT_REQUIRED);
+
+if ($ndbclient_lib) {
+  my $ndbclient_lib_dir = dirname($ndbclient_lib);
+  print $F "--let \$RDRS_NDBCLIENT_LIB= $ndbclient_lib_dir\n"; 
+  print $F "--echo Found RonDB client library\n"
+} else {
+  # Test runs only when ndb is compiled and enabled.
+  # If not able to find libndbclient, probably an issue with this script.
+  print $F "die Did not find ndbclient library\n";
```
Comment (mronstro) [inline mysql-test/suite/rdrs/include/have_rdrs.inc:65]: Isn't this supposed to be followed by a die call?

File: mysql-test/suite/rdrs/include/have_rdrs.inc [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+                                ],
+                                NOT_REQUIRED);
+
+if ($rdrclient_lib) {
+   my $rdrclient_lib_dir = dirname($rdrclient_lib);
+   print $F "--let \$RDRS_CLIENT_LIB= $rdrclient_lib_dir\n";
+   print $F "--echo Found RonDB REST server client library\n"
+} else {
+  print $F "--let \$skip_test=1\n";
+  print $F "--let \$skip_reason=This test requires RonDB REST library\n";
```
Comment (mronstro) [inline mysql-test/suite/rdrs/include/have_rdrs.inc:87]: die call?

File: mysql-test/suite/rdrs/include/have_rdrs.inc [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+                                ],
+                                NOT_REQUIRED);
+
+if ($rdrs_string_lib) {
+  my $rdrs_string_lib_dir = dirname($ndbclient_lib);
+  print $F "--let \$NDBCLIENT_LIB_DIR= $rdrs_string_lib_dir\n";
+  print $F "--echo Found ndbclient library\n"
+} else {
+  print $F "--let \$skip_test=1\n";
+  print $F "--let \$skip_reason=This test requires rdrs_string library\n";
```
Comment (mronstro) [inline mysql-test/suite/rdrs/include/have_rdrs.inc:109]: die call?

File: mysql-test/suite/rdrs/include/have_rdrs.inc [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+$CONF->close();
+
+EOF
+
+#--------------------------------------------------------------------------------
+
+--source $MYSQLTEST_VARDIR/tmp/have_rdrs_result.inc
+
+# skip if the required libs were not found
+if ($skip_test) {
```
Comment (mronstro) [inline mysql-test/suite/rdrs/include/have_rdrs.inc:195]: Not sure how this variable is set/unset

File: mysql-test/suite/rdrs/my.cnf [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+# Set error log verbosity to max, otherwise info messages
+# are filtered away
+log-error-verbosity=3
+# Enable load data local functionality
+local-infile=true
+
+[mysqld.1.1]
+# Specify which nodeids the connection pool on first mysqld should use
+ndb-cluster-connection-pool-nodeids=49,63,127
+# This suite (--suite=ndb) is designed to be run without binlogging,
```
Comment (mronstro) [inline mysql-test/suite/rdrs/my.cnf]: suite=rdrs

File: mysql-test/suite/rdrs/my.cnf [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+# turn off binlogging (and thus also slave updates) for this mysqld
+skip-log-bin
+skip-log-replica-updates
+
+[mysqld.2.1]
+# Specify which nodeids the connection pool on second mysqld should use
+ndb-cluster-connection-pool-nodeids=16,32,48
+# This suite (--suite=ndb) is designed to be run without binlogging,
```
Comment (mronstro) [inline mysql-test/suite/rdrs/my.cnf]: suite=rdrs

File: mysql-test/suite/rdrs/t/batch.test [hunk 7]
```diff
@@ -0,0 +1,15 @@
+#
+# Run the Cluster/J JUnit tests from MTR
```
Comment (mronstro) [inline mysql-test/suite/rdrs/t/batch.test]: RDRS unit tests I presume

File: mysql-test/suite/rdrs/t/mem-buffers.test [hunk 8]
```diff
@@ -0,0 +1,15 @@
+#
+# Run the Cluster/J JUnit tests from MTR
```
Comment (mronstro) [inline mysql-test/suite/rdrs/t/mem-buffers.test]: RDRS unit tests?

File: mysql-test/suite/rdrs/t/pk.test [hunk 9]
```diff
@@ -0,0 +1,15 @@
+#
+# Run the Cluster/J JUnit tests from MTR
```
Comment (mronstro) [inline mysql-test/suite/rdrs/t/pk.test]: RDRS unit tests?

File: mysql-test/suite/rdrs/t/stat.test [hunk 10]
```diff
@@ -0,0 +1,15 @@
+#
+# Run the Cluster/J JUnit tests from MTR
```
Comment (mronstro) [inline mysql-test/suite/rdrs/t/stat.test]: RDRS unit tests?

File: storage/ndb/rest-server/rest-api-server/Makefile [hunk 11]
```diff
@@ -35,8 +35,6 @@ BUILD_COMMAND=go build  -ldflags="-w \
      -X hopsworks.ai/rdrs/version.BRANCH=${BRANCH}" \
      -o ${OUTPUT_DIR}/rdrs ./cmd/server/main.go
 
-LINK_COMMAND=CGO_LDFLAGS="-g -O2 -L${RDRS_LIB_DIR} -lrdrclient"
-
 rdrs:
 	CGO_LDFLAGS="-g -O2 -L${RDRS_LIB_DIR} -lrdrclient" ${BUILD_COMMAND}
```
Comment (mronstro) [inline storage/ndb/rest-server/rest-api-server/Makefile]: is -O2 enough for production builds, often -O3 is used for prod builds?

File: storage/ndb/rest-server/rest-api-server/internal/server/server.go [hunk 12]
```diff
@@ -139,11 +139,20 @@ func (rc *RouterConext) StartRouter() error {
 		}
 	}
 
-	go func() { // Start REST Server
+	httpListner, err := net.Listen("tcp", fmt.Sprintf("%s:%d", rc.RESTServerIP, rc.RESTServerPort))
```
Comment (mronstro) [inline storage/ndb/rest-server/rest-api-server/internal/server/server.go]: spelling s/Listner/Listener

File: storage/ndb/rest-server/rest-api-server/internal/server/server.go [hunk 13]
```diff
@@ -139,11 +139,20 @@ func (rc *RouterConext) StartRouter() error {
...
+	if err != nil {
+		log.Fatalf("HTTP server returned. Error: %v", err)
+	}
+
+	grpcListner, err := net.Listen("tcp", fmt.Sprintf("%s:%d", rc.GRPCServerIP, rc.GRPCServerPort))
```
Comment (mronstro) [inline storage/ndb/rest-server/rest-api-server/internal/server/server.go]: spelling s/grpcListner/grpcListener

**Review by mronstro — APPROVED** [review]
Approved after handling comments

File: mysql-test/suite/rdrs/my.cnf (hunk 5 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/my.cnf]: fixed

File: mysql-test/suite/rdrs/my.cnf (hunk 6 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/my.cnf]: fixed

File: mysql-test/suite/rdrs/t/batch.test (hunk 7 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/t/batch.test]: fixed

File: mysql-test/suite/rdrs/t/mem-buffers.test (hunk 8 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/t/mem-buffers.test]: fixed

File: mysql-test/suite/rdrs/t/pk.test (hunk 9 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/t/pk.test]: fixed

File: mysql-test/suite/rdrs/t/stat.test (hunk 10 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/t/stat.test]: fixed

File: mysql-test/suite/rdrs/include/have_rdrs.inc (hunk 1 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/include/have_rdrs.inc:65]: This is the die command. We write this die command to a file (have_rdrs_result.inc). This file is then imported at the end of this script.

File: mysql-test/suite/rdrs/include/have_rdrs.inc (hunk 2 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/include/have_rdrs.inc:87]: We skip the test if rdrs client lib is not found

File: mysql-test/suite/rdrs/include/have_rdrs.inc (hunk 3 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/include/have_rdrs.inc:109]: same as above, the skip command is written to the have_rdrs_result.inc file. This file is then imported at the end of this script

File: mysql-test/suite/rdrs/include/have_rdrs.inc (hunk 4 above)
Comment (smkniazi) [reply mysql-test/suite/rdrs/include/have_rdrs.inc:195]: this is initialized in file have_rdrs_result.inc

File: storage/ndb/rest-server/rest-api-server/Makefile (hunk 11 above)
Comment (smkniazi) [reply storage/ndb/rest-server/rest-api-server/Makefile]: I have changed it to use 03

File: storage/ndb/rest-server/rest-api-server/internal/server/server.go (hunk 12 above)
Comment (smkniazi) [reply storage/ndb/rest-server/rest-api-server/internal/server/server.go]: fixed

File: storage/ndb/rest-server/rest-api-server/internal/server/server.go (hunk 13 above)
Comment (smkniazi) [reply storage/ndb/rest-server/rest-api-server/internal/server/server.go]: fixed

**Pull request merged**